spring-jdbc/src/main/java/org/springframework/jdbc/datasource/DataSourceTransactionManager.java

```diff
@@ -155,8 +155,8 @@ public DataSourceTransactionManager(DataSource dataSource) {
 	 * @see TransactionAwareDataSourceProxy
 	 * @see org.springframework.transaction.jta.JtaTransactionManager
 	 */
-	public void setDataSource(DataSource dataSource) {
-		if (dataSource instanceof TransactionAwareDataSourceProxy) {
+	public void setDataSource(@Nullable DataSource dataSource) {
+		if (dataSource != null && dataSource instanceof TransactionAwareDataSourceProxy) {
```

Comment:
`instanceof` works fine with `null`, so the extra `!= null` check upfront isn't actually necessary.